src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -48,8 +52,19 @@
      * <p>
      * Inserts an ItemStack into the given slot and return the remainder.
      * The ItemStack <em>should not</em> be modified in this function!
+     * Note: This behaviour is subtly different from {@link IFluidHandler#fill(FluidStack, IFluidHandler.FluidAction)}.
+     * </p>
+     * <p>
+     * insertItem must have the four following properties.
+     * First, the simulation mode must not mutate the inventory.
+     * Second, the simulation mode must be accurate, i.e insertItem(slot, stack, true)
+     * must be equivalent to insertItem(slot, stack, false).
+     * Third, real insertion must be idempotent, i.e insertItem(slot, insertItem(slot, stack, false), false)
```

Comment:
This is not true either, as some item handlers might want to rate-limit the insertion. Here is an example of a **fluid** handler doing it, but I see no reason why item handlers shouldn't be allowed to do the same: https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/c0a9dd43df29f6edce051835bb9d662600497d80/src/main/java/appeng/blockentity/misc/CondenserBlockEntity.java#L268.